lfs.c

```diff
@@ -2514,7 +2514,6 @@ static int lfs_file_rawopencfg(lfs_t *lfs, lfs_file_t *file,
                 {LFS_MKTAG(LFS_TYPE_REG, file->id, nlen), path},
                 {LFS_MKTAG(LFS_TYPE_INLINESTRUCT, file->id, 0), NULL}));
         if (err) {
```

Comment:
Consider change this to `if (err == LFS_ERR_NOSPC) {`

Reply:
I was about to add the following change:

```c
        // it can happen that the file name doesn't fit in the metadata blocks, e.g., a 256 byte file name will
        // not fit in a 128 byte block
        err = (err == LFS_ERR_NOSPC) ? LFS_ERR_NAMETOOLONG : err;
        if (err) {
            goto cleanup;
        }
```

But I'm not sure if that would resolve the issue that I've stumbled upon, since then again I would receive `LFS_ERR_NAMETOOLONG`. My scenario was quite special though in the sense that I've had bugs in my integration (both block device driver and the CRC function, which I've changed to use available CPU instructions). My favourite resolution would simply be a comment documenting your case and return the original error code.